akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/externalvideo/StartExternalVideoPubMsgHdlr.scala

```diff
@@ -0,0 +1,23 @@
+package org.bigbluebutton.core.apps.externalvideo
+
+import org.bigbluebutton.common2.msgs._
+import org.bigbluebutton.core.bus.MessageBus
+import org.bigbluebutton.core.running.{ LiveMeeting }
+
+trait StartExternalVideoPubMsgHdlr {
+
+  def handle(msg: StartExternalVideoPubMsg, liveMeeting: LiveMeeting, bus: MessageBus): Unit = {
+    def broadcastEvent(msg: StartExternalVideoPubMsg) {
+      val routing = Routing.addMsgToClientRouting(MessageTypes.DIRECT, liveMeeting.props.meetingProp.intId, "nodeJSapp")
+      val envelope = BbbCoreEnvelope(StartExternalVideoEvtMsg.NAME, routing)
+      val header = BbbClientMsgHeader(StartExternalVideoEvtMsg.NAME, liveMeeting.props.meetingProp.intId, msg.header.userId)
+
+      val body = StartExternalVideoEvtMsgBody(msg.body.externalVideoUrl)
+      val event = StartExternalVideoEvtMsg(header, body)
+      val msgEvent = BbbCommonEnvCoreMsg(envelope, event)
+      bus.outGW.send(msgEvent)
+    }
+
+    broadcastEvent(msg)
```

Comment:
The message handler needs a PermissionCheck to make sure that only the presenter is the one sending the start message.

Reply:
Totally forgot about this, sorry guys. To include the permission checks we need to make sure whats the behavior of the YouTube playback when the presenter leaves the room. I remember mentioning this with @lfzawacki and I believe this might cause some random user ejections when there is no other moderator in the room to assume the presenter role.

Reply:
I'm not sure where the messages originate in this case, but the way we handle this elsewhere is to have PubMsgs sent by users (or on behalf of) and SysMsgs that are sent from server components and have no trigger for users to generate them. Once that separation is created you can run the permission check against the PubMsg and not against the SysMsg because there's no user sending it.